api/models.py

```diff
@@ -109,7 +109,16 @@ def update(self):
         self.updated = datetime.utcnow()
 
     @classmethod
-    def modify_parent(cls, query_param_dict):
-        """Modify query parameter dictionary for parent"""
-        query_param_dict['parent'] = ObjectId(query_param_dict['parent'])
-        return query_param_dict
+    def translate_fields(cls, params: dict):
+        """Translate fields in `params` into objects as applicable
+
+        Translate fields reprensted by strings in the `params` dictionary into
+        objects that match the model.  For example, database IDs are converted
+        to ObjectId.  Return a new dictionary with the translated values
+        replaced.
+        """
+        translated = params.copy()
```

Comment:
This is to avoid modifying the input argument. I've stopped making functions that do that as they can lead to very subtle issues. Modifying data in-place should be done by methods of an object instead.